File: ContactList.Data/Details.cs
using System.Collections.Generic;

namespace ContactList.Data
{
    public class Details
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Address { get; set; }
        public string JobTitle { get; set; }
        public List<PhoneDTO> Phones { get; set; } 
        public List<EmailDTO> Emails { get; set; }
        public List<TagDTO> Tags { get; set; }
    }
}

File: ContactList/ClientApp/src/app/details.ts
import { Phone } from "./Phone";
import { Email } from "./email";
import { Tag } from "./tag";

export class Details {
  id: number;
  firstName: string;
  lastName: string;
  address: string;
  jobTitle: string;
  phones: Phone[];
  emails: Email[];
  tags: Tag[];

  constructor() {
    this.id = 0;
    this.firstName = "";
    this.lastName = "";
    this.address = "";
    this.jobTitle = "";
    this.phones = [];
    this.emails = [];
    this.tags = [];
  }
}
